client/driver/docker.go

```diff
@@ -893,4 +852,26 @@ func (h *DockerHandle) run() {
 		h.logger.Printf("[ERR] driver.docker: failed to kill the syslog collector: %v", err)
 	}
 	h.pluginClient.Kill()
+
+	// Stop the container just incase the docker daemon's wait returned
+	// incorrectly
+	if err := h.client.StopContainer(h.containerID, 0); err != nil {
+		_, noSuchContainer := err.(*docker.NoSuchContainer)
+		_, containerNotRunning := err.(*docker.ContainerNotRunning)
+		if !containerNotRunning && !noSuchContainer {
+			h.logger.Printf("[ERR] driver.docker: error stopping container: %v", err)
+		}
+	}
+
+	// Remove the container
+	if err := h.client.RemoveContainer(docker.RemoveContainerOptions{ID: h.containerID, Force: true}); err != nil {
+		h.logger.Printf("[ERR] driver.docker: error removing container: %v", err)
+	}
+
+	// Cleanup the image
+	if h.cleanupImage {
+		if err := h.client.RemoveImage(h.imageID); err != nil {
```

Comment:
@dadgar The extra logic's purpose was to only tell the user why an image couldn't be removed. The error from `RemoveImage` API conveys that message anyways and we are not returning this error anyways.